agent/lib/kontena/workers/service_pod_manager.rb

```diff
@@ -62,10 +63,13 @@ def populate_workers_from_master
 
         service_pods.each do |s|
           ensure_service_worker(Kontena::Models::ServicePod.new(s))
+          sleep 0.05
         end
       }
     rescue Kontena::RpcClient::Error => exc
       warn "failed to get list of service pods from master: #{exc}"
+    rescue => exc
+      error exc.message
```

Comment:
What errors is this rescuing? Should it crash and restart instead?

At least log the backtrace.